bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,313 @@
+        config = getConfigAs(PJLinkDeviceConfiguration.class);
+        setupDevice();
+        setupRefreshInterval();
+    }
+
+    private void clearRefreshInterval() {
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    private void handleAuthenticationException(AuthenticationException e) {
+        updateProperty(PJLinkDeviceBindingConstants.PARAMETER_AUTHENTICATION_REQUIRED, new Boolean(true).toString());
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_CONFIGURATION_PENDING, e.getMessage());
+    }
+
+    private void setupDevice() {
+        try {
+            PJLinkDevice device = config.getDevice();
+            device.checkAvailability();
+
+            updateDeviceProperties(device);
+            updateInputChannelStates(device);
+            updateStatus(ThingStatus.ONLINE);
+
+        } catch (ResponseException | IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (AuthenticationException e) {
+            this.handleAuthenticationException(e);
+        }
+    }
+
+    private void setupRefreshInterval() {
+        clearRefreshInterval();
+        boolean atLeastOneChannelToBeRefreshed = PJLinkDeviceHandler.this.config.refreshPower
+                || PJLinkDeviceHandler.this.config.refreshMute || PJLinkDeviceHandler.this.config.refreshInputChannel;
+        if (config.refresh > 0 && atLeastOneChannelToBeRefreshed) {
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refresh, TimeUnit.SECONDS);
+        }
+    }
+
+    private void updateDeviceProperties(PJLinkDevice device) throws IOException, AuthenticationException {
+        updateProperty(PJLinkDeviceBindingConstants.PARAMETER_AUTHENTICATION_REQUIRED,
+                device.getAuthenticationRequired().toString());
+
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_NAME, device.getName());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_NAME, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MANUFACTURER, device.getManufacturer());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MANUFACTURER, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MODEL, device.getModel());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MODEL, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_COMBINED_ID, device.getFullDescription());
```

Comment:
Is there a real interest to provide two different properties, PARAMETER_NAME and PARAMETER_COMBINED_ID ?

Reply:
I thought it was handy for diagnosis, but I suppose the users can also copy three fields (manufacturer, model and name). I have removed the property.